packages/grid/_modules_/grid/components/containers/GridRoot.tsx

```diff
@@ -6,7 +6,7 @@ import { GridRootPropsContext } from '../../context/GridRootPropsContext';
 import { optionsSelector } from '../../hooks/utils/optionsSelector';
 import { GridRootContainerRef } from '../../models/gridRootContainerRef';
 import { useStyles } from './GridRootStyles';
-import { visibleGridColumnsLengthSelector } from '../../hooks/features/columns/gridColumnsSelector';
+import { visibleGridColumnsLengthSelector } from '../../hooks/features/columnVisibility/gridVisibleColumnsSelector';
```

Comment:
As you suggested it makes more sense to keep the new hook under `/columns` as well.

On another note. I was about to ask why did we name the selector `visibleGridColumnsLengthSelector` but quick git blame shows that I am responsible for this 😅. Now that I think about it to keep it consistent with the other selectors maybe we should rename it to `gridVisibleColumnsLengthSelector`?

Reply:
:smile: yes no problem for renaming it

If we keep both under the same folder, would you still move the methods like `getColumnsMeta`, `getColumnPosition` in `useGridVisibleColumns`

Basically, should `useGridVisibleColumns` handle everything that is related to the visible columns or just the hide / show  behavior ?

Reply:
Mmm, these methods sound like something that is generic for all columns regardless if it is visible or not. I would probably keep them in the `useGridColumns` but let's see what the others have to say about it. I don't really have a strong personal preference.